third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/WSCodePartition.cpp

```diff
@@ -1529,26 +1532,24 @@ desyncTCGen5MMAOp(OpBuilderWithAsyncTaskIds &builder, ttng::TCGen5MMAOp mmaOp,
   // is false this wait_barrier serves as consumer_wait.
   builder.setInsertionPoint(producerOrConsumer);
   builder.setAsyncTaskIdsFromOp(producerOrConsumer);
-  auto producerBarrier = getBarrierForPipelineStage(
-      builder, barrierAlloc, bufferIdx, producerOrConsumer);
+  builder.setLoopScheduleInfo(producerOrConsumer);
+  auto producerBarrier =
+      getBarrierForPipelineStage(builder, barrierAlloc, bufferIdx);
   // curPhase = curPhase xor True for emptyBarrier.
   Value phase = inPhase;
   auto loc = producerOrConsumer->getLoc();
   if (asProducerAcquire) {
     Value _1_1b =
         builder.createWithAsyncTaskIds<arith::ConstantIntOp>(loc, 1, 1);
-    copyLoopScheduleInfo(_1_1b.getDefiningOp(), producerOrConsumer);
     // Creating phase for producerOrConsumer.
     phase = builder.createWithAsyncTaskIds<mlir::arith::XOrIOp>(loc, inPhase,
                                                                 _1_1b);
-    copyLoopScheduleInfo(phase.getDefiningOp(), producerOrConsumer);
   }
   phase = builder.createWithAsyncTaskIds<arith::ExtSIOp>(
       loc, builder.getI32Type(), phase);
-  copyLoopScheduleInfo(phase.getDefiningOp(), producerOrConsumer);
   auto waitOp = builder.createWithAsyncTaskIds<ttng::WaitBarrierOp>(
       loc, producerBarrier, phase);
-  copyLoopScheduleInfo(waitOp, producerOrConsumer);
+  builder.clearLoopScheduleInfo();
```

Comment:
I opted to clear whenever I saw the builder returned unless AsyncTasks restored the old tasks. In those cases I restored the loop schedule for consistency.